Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -81,55 +135,89 @@ Quality RawDataReaderCheck::check(std::map<std::string, std::shared_ptr<MonitorO
           vGoodBC.push_back(i);
         }
       }
-      if (vBadBC.size() > 0)
+      if (vBadBC.size() > 0) {
         result = Quality::Bad;
-      else if (vMediumBC.size() > 0)
+      } else if (vMediumBC.size() > 0) {
         result = Quality::Medium;
-      else
+      } else {
         result = Quality::Good;
-    }
-    if (mo->getName() == "inputs" || mo->getName() == "classes") {
+      }
+    } else {
       auto* h = dynamic_cast<TH1F*>(mo->getObject());
-      if (mo->getName() == "inputs" && (!fHistInputPrevious || inputCycleCounter < 2)) {
-        fHistInputPrevious = (TH1F*)h->Clone();
-        inputCycleCounter++;
-      } else if (mo->getName() == "classes" && (!fHistClassesPrevious || classCycleCounter < 2)) {
-        fHistClassesPrevious = (TH1F*)h->Clone();
-        classCycleCounter++;
-      } else {
+      std::string moName = mo->getName();
+      if (moName.find("Ratio") != std::string::npos) {
+        relativeRates = true;
+      }
+      if (moName.find("input") != std::string::npos) {
+        inputRates = true;
+      }
+      if (relativeRates && inputRates) {
+        h->Scale(1. / h->GetBinContent(3));
```

Comment:
FYI, this modifies the histogram you are checking, it will be stored in QCDB like that. Perhaps you may want to move modification to `beautify()` for clarity, but if this is what you want, it's OK. Please keep in mind though, that if you ever apply more than one check on this object, the 2nd one will get the modified MO.

Reply:
This is what we need to do. In this case, we want to check how the ratio between inputs and the specified minimum bias input changed after each cycle. As the bins represent different inputs, we need to scale the histogram first and then perform the check. There is only one check performed on the histogram.